dashboard/app/views/levels/editors/_droplet.html.haml

```diff
@@ -1,23 +1,36 @@
-.field
-  = f.label :edit_code,'Droplet level'
-.field
-  = f.label :code_functions, 'Edit Droplet Palette'
-  = f.text_area :code_functions, placeholder: 'Droplet Palette', rows: 4, value: @level.code_functions ? JSON.pretty_generate(@level.code_functions) : ''
-
-.field
-  = f.label :start_blocks, 'Edit Starting JavaScript Program'
-  .row
-    .span8
-      ~ f.text_area :start_blocks, placeholder: 'Start Blocks', rows: 4, value: @level.uses_droplet? ? @level.start_blocks : @level.pretty_block('start')
-    #start-preview.span4
-.field
-  = f.label :palette_category_at_start
-  = f.select :palette_category_at_start, options_for_select(@level.class.palette_categories, @level.palette_category_at_start), {include_blank: true}
-.field
+%legend.control-legend.collapsed{data: {toggle: "collapse", target: "#toolbox"}}
+  Toolbox
+
+#toolbox.in.collapse
   = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :droplet_tooltips_disabled, description: "Hide droplet documentation tooltips"}
-.field
   = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :lock_zero_param_functions, description: "Don't allow parameters to be added to user-defined functions with no parameters"}
-.field
+
+  - if @level.is_a? Applab
+    = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :makerlab_enabled, description: "Enable Maker APIs"}
+    = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :autocomplete_palette_apis_only, description: "Autocomplete palette APIs only"}
+    = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :execute_palette_apis_only, description: "Execute palette APIs only"}
+    = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :beginner_mode, description: "Beginner mode for loops"}
```

Comment:
it's a little weird to see fields pulled out of an applab-specific partial, only to be put behind an applab conditional. but it's more important to arrange the fields the way that the curriculum team wants than it is to keep conditionals out of our code, so this still seems like an improvement to me.

Reply:
I agree. I'm kinda trying to organize in small chunks that make sense and I think there will be some moving things to possibly move them again in order to get this right. The mistake I made last time was trying to make it right on the first go and that was too hard to follow for reviews.

Reply:
No worries, makes sense! FWIW the most vocal feedback I recall receiving last time was from Elizabeth objecting to having things being hidden by default, since that broke her flow of doing a text search to find the field that she was looking for. I think you're doing a fine job of avoiding that here by making all the expand/collapse controls be expanded by default.